requirements/dev.txt

```diff
@@ -1,11 +1,13 @@
 #
-# This file is autogenerated by pip-compile with Python 3.11
+# This file is autogenerated by pip-compile with Python 3.12
```

Comment:
... why does the script include the python version in here? Is it likely to be give different output depending on which python version used?
Anyway vcpkg is using python 3.11 right now so I've been using that everywhere - including to generate these files - in case it matters. But I suspect it doesn't

Reply:
I think for what we're doing it most likely doesn't... but I wonder if we can encourage CMake to pick up Py3.11 if it's available during config.